Misc/NEWS.d/next/Core and Builtins/2022-03-19-21-47-36.bpo-44800.XnGKoD.rst

```diff
@@ -0,0 +1,5 @@
+Renamed ``_PyInterpreterFrame`` as ``_Py_frame`` to emphasise its close
+association with ``PyFrameObject`` (they're conceptually the same thing, but
```

Comment:
What does "conceptually the same thing" mean?
They aren't the same thing. What is the concept that makes them the same?

Reply:
The same thing that makes it OK to pass both `PyFrameObject ` and `_PyInterpreterFrame` structs to functions that have names starting with `_PyFrame`: they're Python code evaluation frames either way, but you may be working with the underlying data storage directly, or you may be working with the full refcounted Python object.

The difference between the two is mechanical (how their memory allocation is managed) rather than conceptual.

Reply:
Your concepts are just that, *your* concepts. They are highly subjective.

`PyFrameObject`s and `_PyInterpreterFrame` are different things. One is a reference counted heap object, the other a stack frame. They have to managed quite differently, to me that means they are "conceptually distinct".